### PR TITLE
feat(manager-components): add <Region /> component

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-15076
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
Component must have 2 modes:
1. [default]mode=region. name looks like eu-west-par
2. mode=datacenter. name looks like SBG

Add < Region / > component

< Region mode="region" name="eu-west-par" / >
Output: "Europe (France - Paris)"

or

< Region mode="datacenter" name="SBG" code="5" / >
Output: "Strasbourg (SBG5) - France"

### CODE REVIEW

File: packages/manager-components/src/components/region-label/region-label.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import RegionLabel, { RegionLabelProps } from './region-label.component';
+import { DemoRegion, DemoDatacenter } from './region-label.stories';
+import translatedRegion from './translations/Messages_fr_FR.json';
+
+const renderComponent = (props: RegionLabelProps) => {
+  return render(<RegionLabel {...props} />);
+};
+
+describe('Region Label component', () => {
+  it('renders region label correctly for region key', async () => {
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager-components/src/components/region-label/region-label.spec.tsx:12]:
```suggestion
  it('renders region label correctly for region key', async () => {
```
  it('renders region label correctly for region key', async () => {
    renderComponent({
      ...DemoRegion.args,
    });
    const regionKey = DemoRegion.args.name
      .replace(/-/g, '_')
      ?.toLocaleLowerCase();
    const regionLabelElement = screen.getByText(
      translatedRegion[`region_${regionKey}`],
    );
    expect(regionLabelElement).toBeVisible();
  });

  it('renders region label correctly for datacenter key', async () => {
    renderComponent({
      ...DemoDatacenter.args,
    });
    const regionKey = DemoDatacenter.args.name
      .replace(/-/g, '_')
      ?.toLocaleLowerCase();
    const regionLabelElement = screen.getByText('Roubaix (RBX2) - France');
    expect(regionLabelElement).toBeVisible();
  });

File: packages/manager-components/src/components/region-label/region-label.component.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useTranslation } from 'react-i18next';
+import './translations';
+
+export interface RegionLabelProps {
+  mode: string;
+  name: string;
+  micro: number;
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager-components/src/components/region-label/region-label.component.tsx:7]:
```suggestion
  micro?: number;
```

File: packages/manager-components/src/components/region-label/region-label.component.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export function RegionLabel({
+  mode = 'region',
+  name,
+  micro,
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager-components/src/components/region-label/region-label.component.tsx:13]: Or by Default put a number

Comment (sachinrameshn) [reply packages/manager-components/src/components/region-label/region-label.component.tsx:13]: Its done Alex. Thanks !!

File: packages/manager-components/src/components/region-label/translations/Messages_fr_FR.json
Comment (anooparveti) [inline packages/manager-components/src/components/region-label/translations/Messages_fr_FR.json:1]: **nitpick:** Can we split this into 2 namespaces to facilitate loading based on the `mode`?

File: packages/manager-components/src/components/region-label/region-label.component.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { useTranslation } from 'react-i18next';
+import './translations';
+
+export interface RegionLabelProps {
+  mode: string;
+  name: string;
+  micro?: number;
+}
+
+export function RegionLabel({
```
Comment (anooparveti) [inline packages/manager-components/src/components/region-label/region-label.component.tsx:10]:
```suggestion
export function Region({
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager-components/src/components/region-label/region-label.component.tsx (hunk 4 above)
Comment (sachinrameshn) [reply packages/manager-components/src/components/region-label/region-label.component.tsx]: done Anoop

File: packages/manager-components/src/components/region-label/translations/Messages_fr_FR.json
Comment (sachinrameshn) [reply packages/manager-components/src/components/region-label/translations/Messages_fr_FR.json:1]: Its done Anoop

File: packages/manager-components/src/components/region/region.component.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+export interface RegionProps {
+  mode: string;
+  name: string;
+  micro?: number;
+}
+
+export function Region({ mode = 'region', name, micro }: RegionProps) {
+  const { t: tRegion } = useTranslation('region');
+  const { t: tDatacenter } = useTranslation('datacenter');
```
Comment (anooparveti) [inline packages/manager-components/src/components/region/region.component.tsx:13]: **nitpick:** depending on the mode, you only need to load one among these.
```suggestion
  const { t } = useTranslation(mode === 'region' ? 'region' : 'datacenter');
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager-components/src/components/region/region.component.tsx (hunk 5 above)
Comment (sachinrameshn) [reply packages/manager-components/src/components/region/region.component.tsx:13]: Yes you are right or else it loads both. I will fix it.

File: packages/manager-components/src/components/region/region.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+describe('Region component', () => {
+  it('renders region correctly for region key', async () => {
+    renderComponent({
+      ...DemoRegion.args,
+    });
+    const regionKey = DemoRegion.args.name
+      .replace(/-/g, '_')
+      ?.toLocaleLowerCase();
```
Comment (anooparveti) [inline packages/manager-components/src/components/region/region.spec.tsx:18]: **question:** Are you sure this test-case is passing?

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager-components/src/components/region/region.spec.tsx (hunk 6 above)
Comment (sachinrameshn) [reply packages/manager-components/src/components/region/region.spec.tsx]: Its corrected now.

**Review by anooparveti — DISMISSED** [review]

File: packages/manager-components/src/components/region/region.component.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { useTranslation } from 'react-i18next';
+import './translations/region';
+import './translations/datacenter';
+
+export interface RegionProps {
+  mode: string;
```
Comment (chipp972) [inline packages/manager-components/src/components/region/region.component.tsx:6]:
```suggestion
  mode: 'region' | 'datacenter';
```

**Review by chipp972 — DISMISSED** [review]

File: packages/manager-components/src/components/region/region.component.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  name: string;
+  micro?: number;
+}
+
+export function Region({ mode = 'region', name, micro }: RegionProps) {
+  const { t } = useTranslation(mode === 'region' ? 'region' : 'datacenter');
+
+  return (
+    <>
+      {mode === 'region' ? t(`region_${name}`) : t(`region_${name}`, { micro })}
```
Comment (chipp972) [inline packages/manager-components/src/components/region/region.component.tsx:16]:
```suggestion
      {t(`region_${name}`, { micro })}
```

passing `micro: undefined` won't affect the translation in the case of region

File: packages/manager-components/src/components/region/region.component.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import './translations/region';
+import './translations/datacenter';
+
+export interface RegionProps {
+  mode: 'region' | 'datacenter';
+  name: string;
+  micro?: number;
+}
+
+export function Region({ mode = 'region', name, micro }: RegionProps) {
```
Comment (aboungnaseng-ovhcloud) [inline packages/manager-components/src/components/region/region.component.tsx:11]: there is no micro value by default ?

**Review by anooparveti — APPROVED** [review]

File: packages/manager-components/src/components/region/region.component.tsx (hunk 9 above)
Comment (sachinrameshn) [reply packages/manager-components/src/components/region/region.component.tsx:11]: nope

**Pull request merged**